### PR TITLE
LNK-3053: Added the option to select a daily POI list (as well as oth…

### PR DESCRIPTION
**Pull request by ajagann1**
…er frequencies) instead of defaulting to monthly POI lists

LNK-3058: Added Today and Yesterday as options to be selected for reporting period
LNK-3059: Added ReportingPeriodMethods to EhrPatientList to allow selection of reporting period lengths within tenant config

Combined these tickets together due to the changes being directly related to one another.

## Summary by CodeRabbit

- **New Features**
	- Enhanced patient list handling with flexible reporting period calculations.
	- New reporting period options: "Yesterday" and "Today" for improved reporting.
	- Streamlined patient ID processing in patient list management.
	- Added validation for tenant-specific time zone configurations in report generation.

- **Bug Fixes**
	- Improved error handling for missing configurations in report generation.

- **Documentation**
	- Updated logging for better traceability in patient list conversions and report generation.

### CODE REVIEW

File: api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java [hunk 1]
```diff
@@ -93,7 +92,8 @@ public void queryPatientList(@PathVariable String tenantId) throws Exception {
       }
 
       for (int j = 0; j < list.getMeasureId().size(); j++) {
-        PatientList patientList = this.convert(tenantService, sources, list.getMeasureId().get(j), new URI(client.getServerBase()));
+        PatientList patientList = this.convert(tenantService, sources, list.getMeasureId().get(j),
+                new URI(client.getServerBase()), list.getReportingPeriodMethod());
```
[the remark this answers is not in the document]
Comment (ajagann1) [reply api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java:96]: Not necessary here.  The EhrPatientList class, which is where the ReportingPeriodMethod is pulling from, defaults it to CurrentMonth and only overwrites that with what's in the config if there's something there.

File: api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java [hunk 2]
```diff
@@ -155,15 +155,15 @@ private ListResource readList(IGenericClient client, String patientListId) {
...
+                              ReportingPeriodMethods reportingPeriodMethod) throws URISyntaxException {
     logger.info("Converting List resources to DB PatientLists");
     PatientList patientList = new PatientList();
 
     var timeZoneId = tenantService.getConfig().getTimeZoneId();
     var timeZone = TimeZone.getTimeZone(Objects.requireNonNullElse(timeZoneId, ZoneId.systemDefault().getId()));
 
-    // TODO: Make ReportingPeriodMethods configurable
-    ReportingPeriodCalculator calculator = new ReportingPeriodCalculator(ReportingPeriodMethods.CurrentMonth, timeZone);
+    ReportingPeriodCalculator calculator = new ReportingPeriodCalculator(reportingPeriodMethod, timeZone);
```
[the remark this answers is not in the document]
Comment (ajagann1) [reply api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java:166]: Same as above, should never be null.

File: api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java [hunk 3]
```diff
@@ -155,15 +155,15 @@ private ListResource readList(IGenericClient client, String patientListId) {
     return list;
   }
 
-  private PatientList convert(TenantService tenantService, List<ListResource> sources, String identifier, URI baseUrl) throws URISyntaxException {
+  private PatientList convert(TenantService tenantService, List<ListResource> sources, String identifier, URI baseUrl,
+                              ReportingPeriodMethods reportingPeriodMethod) throws URISyntaxException {
```
[the remark this answers is not in the document]
Comment (ajagann1) [reply api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java:159]: Same as above

File: api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java [hunk 4]
```diff
@@ -32,6 +37,9 @@ public class GenerateAndSubmitReportTask implements Runnable {
   @Autowired
   private ReportController reportController;
 
+  @Autowired
+  private SharedService sharedService;
```
[the remark this answers is not in the document]
Comment (ajagann1) [reply api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java:42]: Using a constructor for this class would interfere with bean initialization in the Scheduler class.

File: api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java [hunk 5]
```diff
@@ -32,6 +38,13 @@ public class GenerateAndSubmitReportTask implements Runnable {
...
+
+  private boolean validTimeZone(String timezone) {
+    return Set.of(TimeZone.getAvailableIDs()).contains(timezone);
+  }
```
[the remark this answers is not in the document]
Comment (ajagann1) [reply api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java]: Done

**Review by smailliwcs — APPROVED** [review]

File: api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java [hunk 6]
```diff
@@ -44,8 +52,13 @@ public void run() {
       throw new IllegalArgumentException("reportingPeriodMethod");
     }
 
+    TenantService tenantService = TenantService.create(this.sharedService, tenantId);
+    TimeZone timezone = TimeZone.getTimeZone(
+            Objects.requireNonNullElse(tenantService.getConfig().getTimeZoneId(),
+                    ZoneId.systemDefault().getId()));
+
```
[the remark this answers is not in the document]
Comment (smailliwcs) [reply api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java]: Resolving this conversation, though we could certainly add this to the validations we perform in `TenantController.create`/`updateTenant`.

File: api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java [hunk 7]
```diff
@@ -44,8 +52,13 @@ public void run() {
       throw new IllegalArgumentException("reportingPeriodMethod");
     }
 
+    TenantService tenantService = TenantService.create(this.sharedService, tenantId);
+    TimeZone timezone = TimeZone.getTimeZone(
+            Objects.requireNonNullElse(tenantService.getConfig().getTimeZoneId(),
+                    ZoneId.systemDefault().getId()));
```
[the remark this answers is not in the document]
Comment (smailliwcs) [reply api/src/main/java/com/lantanagroup/link/api/scheduling/GenerateAndSubmitReportTask.java]: As above, consider time zone validation as a future enhancement in `TenantController`.

**Pull request merged**